### PR TITLE
Difftest core: small PR removes some todos

### PR DESCRIPTION
**Pull request by danwt**
# Description

Removes some TODOs, clarifies one

## Type of change

Refactor

### CODE REVIEW

File: tests/difference/core/driver/core_test.go [hunk 1]
```diff
@@ -237,7 +237,6 @@ func (s *CoreSuite) matchState() {
 		for j := 0; j < initState.NumValidators; j++ {
 			s.Require().Equalf(int64(s.traces.Tokens(j)), s.providerTokens(int64(j)), diagnostic+"P tokens mismatch for val %d", j)
 		}
-		// TODO: delegations
```
Comment (danwt) [inline tests/difference/core/driver/core_test.go:240]: Not a good idea/ not worth doing. The method is blackbox and delegations are implementation detail.
More docs/ onboarding material is coming which will explain method and reasoning.

File: tests/difference/core/driver/core_test.go [hunk 2]
```diff
@@ -256,7 +255,6 @@ func (s *CoreSuite) matchState() {
 				s.Require().Errorf(err, diagnostic+" power mismatch for val %d, expect 0 (nil), got %d", j, actual)
 			}
 		}
-		// TODO: outstanding downtime
```
Comment (danwt) [inline tests/difference/core/driver/core_test.go:259]: Not a good idea/ not worth doing. The method is blackbox and delegations are implementation detail.
More docs/ onboarding material is coming which will explain method and reasoning.

File: tests/difference/core/driver/core_test.go [hunk 3]
```diff
@@ -313,7 +311,11 @@ func (s *CoreSuite) TestAssumptions() {
 		s.T().Fatal(FAIL_MSG)
 	}
 
-	// TODO: write assumption that checks that throttle params are appropriate
+	// TODO: Write a check to make sure that the slash throttle params are set correctly.
+	// 		 The params should be set such that the slash throttle never kicks in and stop a slash.
+	// 		 This is because the model assumes that a slash will always be executed, no matter
+	// 		 how many. This can be achieve by setting the slash factor to e.g. 1.0 and the refresh
+	// 		 period to 1 block.
```
Comment (danwt) [inline tests/difference/core/driver/core_test.go:318]: I'll try to get to this but it's very low prio so I enhanced the description.

File: tests/difference/core/model/src/model.ts [hunk 4]
```diff
@@ -7,7 +7,6 @@ import { strict as assert } from 'node:assert';
  * This model may need updating pending
  * https://github.com/cosmos/ibc/issues/825 (model updated, spec has open PR)
  * https://github.com/cosmos/ibc/issues/796 (model updated, spec awaiting PR)
- * TODO: implement automatic commit hash comparison warning against spec
```
Comment (danwt) [inline tests/difference/core/model/src/model.ts:10]: Not worth doing/ too much hassle.

**Review by mpoke — APPROVED** [review]
Thanks @danwt

**Review by smarshall-spitzbart — APPROVED** [review]
LGTM thanks Dan!

**Pull request merged**